frontend/src/javascript/codeParser/CodeParser.js

```diff
@@ -128,13 +129,22 @@ class CodeUtils {
     }
 
     static removeVariableTags(code) {
-        code = code.replaceAll("<algodebug-variable>", "").replaceAll("</algodebug-variable>", "");
+        let i = -1;
+        while ((i = code.indexOf("<algodebug-variable>")) > 0) {
+            let start = i + "<algodebug-variable>".length;
+            let end = code.indexOf("</algodebug-variable>", start);
+            let variableID = code.substring(start, end);
+            let variableName = variableID.split("@")[0];
+            code = code.substring(0, i) + variableName + code.substring(end + "</algodebug-variable>".length);
+        }
         return code;
     }
 
     static replaceBreakpointTags(code, parsedBreakpoints) {
         for (let breakpoint of parsedBreakpoints) {
-            let variables = breakpoint.variables.map((variable) => `ALGODEBUG_VARIABLE(${variable.name})`).join(" << ");
+            let variables = breakpoint.variables
+                .map((variable) => `ALGODEBUG_VARIABLE(${variable.name}, ${variable.id.split("@")[1]})`)
```

Comment:
Trzeba to rozdzielać? Nie można po prostu variable.id?

Reply:
Pytasz o ogólne rozdzielenie na dwa argumenty? Trzeba. Define musi znać dosłowną nazwę zmiennej, bo potem odnosząc się do niej ją cout-uje. 
Czy drugi argument też trzeba rozdzielać? Niby nie, ale wtedy kod debugujący wygląda: `ALGODEBUG_VARIABLE(nazwa, "nazwa@69")`, co wydaje mi się trochę zbędną redundancją względem `ALGODEBUG_VARIABLE(nazwa, 69)`